trivial - add catch

jshaughn opened the pull request:
@mtho11 just adding that missing catch you noticed.

mtho11 commented:
Can you hook this into showing the error on screen with `setState` and `alertVisible`

jshaughn commented:
@mtho11 , updated with [I think] what you want.  By the way, it would be nicer if we added more to the errors returned from API issues.  I tested this to force a 404 and the message was not much more than, "got a 404", appending the request URL would be useful.  Currently above my pay scale but something to look at doing.
